pkg/hub/lease/controller.go

```diff
@@ -0,0 +1,180 @@
+package lease
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	clientset "github.com/open-cluster-management/api/client/cluster/clientset/versioned"
+	clusterv1informer "github.com/open-cluster-management/api/client/cluster/informers/externalversions/cluster/v1"
+	clusterv1listers "github.com/open-cluster-management/api/client/cluster/listers/cluster/v1"
+	clusterv1 "github.com/open-cluster-management/api/cluster/v1"
+	"github.com/open-cluster-management/registration/pkg/helpers"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	coordv1 "k8s.io/api/coordination/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	coordinformers "k8s.io/client-go/informers/coordination/v1"
+	coordlisters "k8s.io/client-go/listers/coordination/v1"
+)
+
+const leaseDurationTimes = 5
+
+// leaseController checks the lease of managed clusters on hub cluster to determine a managed cluster whether is available.
+type leaseController struct {
+	clusterClient   clientset.Interface
+	clusterLister   clusterv1listers.ManagedClusterLister
+	leaseLister     coordlisters.LeaseLister
+	clusterLeaseMap *clusterLeaseMap
+}
+
+// NewClusterLeaseController creates a cluster lease controller on hub cluster.
+func NewClusterLeaseController(
+	clusterClient clientset.Interface,
+	clusterInformer clusterv1informer.ManagedClusterInformer,
+	leaseInformer coordinformers.LeaseInformer,
+	recorder events.Recorder) factory.Controller {
+	c := &leaseController{
+		clusterClient:   clusterClient,
+		clusterLister:   clusterInformer.Lister(),
+		leaseLister:     leaseInformer.Lister(),
+		clusterLeaseMap: newClusterLeaseMap(),
+	}
+	return factory.New().
+		WithInformers(clusterInformer.Informer(), leaseInformer.Informer()).
+		WithSync(c.sync).
+		ResyncEvery(1*time.Minute).
```

Comment:
This looks like a time to extend the factory with the ability to provide your own resync functionality.  I think you want items in the workqueue to be named for the particular spoke clusters you're investigating.